packages/react-router-dev/vite/plugin.ts

```diff
@@ -309,6 +309,28 @@ const getRouteModuleExports = async (
   ctx: ReactRouterPluginContext,
   routeFile: string,
   readRouteFile?: () => string | Promise<string>
+): Promise<string[]> => {
+  let routePath = path.resolve(ctx.reactRouterConfig.appDirectory, routeFile);
+  let code = await (readRouteFile?.() ?? fse.readFile(routePath, "utf-8"));
+
+  // temporary back compat
```

Comment:
Note to self: should probably create a "remove back compat" and add links to that in these comments so its easier to track these down later.